### PR TITLE
fix(ScrollMenu): prevent mousedown on part with scrollbar

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-2668]

### Description

Move the prevent mouseDown to the div with the scrollbar to be able to use search input in fixedHeader

### How to test

- TagSelector and Autocomplete do not close options when using the scrollbar
- search input in the Select component is focusable

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure that tests pass by running `yarn test`
- [x] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

**Breaking change**

- codemod is created and showcased in the changeset
- test alpha package of Picasso in StaffPortal

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-2668]: https://toptal-core.atlassian.net/browse/FX-2668?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by OndrejTuma — APPROVED** [review]
Tested. Works as expected 💪

File: packages/picasso/src/ScrollMenu/test.tsx [hunk 1]
```diff
@@ -105,4 +105,30 @@ describe('ScrollMenu', () => {
...
+    const footer = getByTestId('footer')
+
+    const isListPrevented = !fireEvent.mouseDown(list)
+    const isFooterPrevented = !fireEvent.mouseDown(footer)
+    const isHeaderNotPrevented = fireEvent.mouseDown(header)
+
+    expect(isListPrevented).toBeTruthy()
+    expect(isFooterPrevented).toBeTruthy()
+    expect(isHeaderNotPrevented).toBeTruthy()
+  })
```
Comment (dmaklygin) [inline packages/picasso/src/ScrollMenu/test.tsx:133]: 👍

**Review by dmaklygin — APPROVED** [review]
Good work!

File: packages/picasso/src/ScrollMenu/ScrollMenu.tsx [hunk 2]
```diff
@@ -49,6 +53,11 @@ export const scrollToSelection = (
   }
 }
 
+const preventClick = (e: React.MouseEvent) => {
+  // ScrollMenu is used in dropdowns. Prevents blur --> dropdown close when scrolled.
```
Comment (LashaJini) [inline packages/picasso/src/ScrollMenu/ScrollMenu.tsx:57]: Could you explain what you mean in second sentence, please?

**Review by LashaJini — APPROVED** [review]
Tested locally :+1:

File: packages/picasso/src/ScrollMenu/ScrollMenu.tsx [hunk 3]
```diff
@@ -49,6 +53,12 @@ export const scrollToSelection = (
   }
 }
 
+const preventClick = (e: React.MouseEvent) => {
+  // ScrollMenu is used in dropdowns.
+  // When clicking on the scrollView, the dropdown should not close.
```
Comment (LashaJini) [inline packages/picasso/src/ScrollMenu/ScrollMenu.tsx:58]: :+1:

File: packages/picasso/src/ScrollMenu/ScrollMenu.tsx [hunk 4]
```diff
@@ -59,32 +69,39 @@ const ScrollMenu = (props: Props) => {
...
       {fixedHeader}
-      <div ref={menuRef} className={classes.scrollView} onBlur={onBlur}>
-        {children}
+      <div onMouseDown={preventClick}>
+        <div
+          data-testid={testIds?.list}
+          ref={menuRef}
+          className={classes.scrollView}
+          onBlur={onBlur}
+          onMouseDown={preventClick}
```
Comment (OndrejTuma) [inline packages/picasso/src/ScrollMenu/ScrollMenu.tsx:99]: you can remove this line 👍

Comment (TomasSlama) [reply packages/picasso/src/ScrollMenu/ScrollMenu.tsx]: ✅

**Review by OndrejTuma — APPROVED** [review]
You were right, a lot of snapshot changes 😬 
But it makes sense to me 👍 
The rest is good and works as expected 🎉

**Review by separatio — COMMENTED** [review]
While overall this looks good, you've introduced 1 new bug.
Clicking the outline of the Input jumps the focus from the Input to the Select component, and back. 

The other issue of opening and closing of the Select component behavior already exists on master.
Please check.

Peek 2022-05-03 12-26

**Review by separatio — APPROVED** [review]
Since https://toptal-core.atlassian.net/browse/FX-2703 was created, we are :ok: with this PR. :shipit:

**Pull request merged**